src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -428,6 +449,91 @@ CitusModifyBeginScan(CustomScanState *node, EState *estate, int eflags)
 	 * locks as early as possible.
 	 */
 	ReacquireMetadataLocks(taskList);
+
+	/*
+	 * If we deferred shard pruning to the executor, then we still need to assign
+	 * shard placements. We do this after acquiring the metadata locks to ensure
+	 * we can't get stale metadata. At some point, we may want to load all
+	 * placement metadata here.
+	 */
+	if (deferredPruning)
+	{
+		AssignTaskPlacements(workerJob);
```

Comment:
Can't this just be: `workerJob->taskList = FirstReplicaAssignTaskList(workerJob->taskList)`?